src/sysen3/papsav.1

```diff
@@ -0,0 +1,397 @@
+
+	; Source reconstructed by EJS on 2018-12-15 from CHANNA;RAKASH PAPSAV binary
+	; Original PAPSAV inserted SYSENG; CALRET >. Reconstructed version does not, 
+	; due to issues with the .CALL macro. The original PAPSAV shared some code
+	; with SYSEN1;SYSMSG > (which also inserted SYSENG; CALRET >). Where applicable,
+	; the source from SYSMSG was used in this reconstruction.
+	
+t1=1
+t=2
+tt=3
+a=4
+.prsva=4
+b=5
+c=6
+f=7
+pt=10
+ch=11
+count=12
+.prsvz=13
+
+;;; Registers Used Internally by the System
+
+P=17		;PushDown List Pointer
+.CF.=16		;Current Stack Frame Pointer
+.PF.=15		;Previous Stack Frame Pointer
+.AP.=14		;Argument List Pointer
+
+TYIC==2			; TTY INPUT CHANNEL
+TYOC==1			; tty output channel
+TYO=<.IOT TYOC,>	; INSTRUCTION TO OUTPUT A CHAR
+
+;;; MAIN PROGRAM
+
+i==4
+l==5
+v==6
+lim==7
+
+	loc 100
+
+;;; Entry Operator
+
+.ENTR.: MOVEI TT,1(P)		;save registers between frames
+	HRLI TT,.PRSVA
+	ADD P, [.PF.-.PRSVA+1,,.PF.-.PRSVA+1] ;protect reg save area
+	BLT TT,0(P)
+	MOVE .AP.,T1		;set called routine's registers
+	MOVE .PF.,.CF.
+	MOVE .CF.,P
+	ADD P,0(T)		;adjust stack
+	JUMPL P,1(T)		;and go to called routine
+	.SUSET [.SIPIRQ,,[%PIPDL]]
+	JRST 1(T)
+
+;;; Return Operator
+
+.RETN.:	MOVE P,.CF.		;pop frame off
+	MOVE .CF.,.PF.
+	MOVSI TT, -<.PF.-.PRSVA>(P)  ;restore registers
+	HRRI TT,.PRSVA
+	BLT TT,.PF.
+	SUB P, [.PF.-.PRSVA+1,,.PF.-.PRSVA+1] ;pop regs off
+	POPJ P,			;and return
+
+CONSTANTS
+
+	0
+FMTBL:	0			;0 end of message
+	TYPOCT			;1 full word octal
+	TYPDEC			;2 full wd decimal
+	TYPHAF			;3 " with commas
+	TYCRLF			;4 do cr
+	CPOPJ			;5 ?
+	TYPSIX			;6 sixbit
+	TYPASZ			;7 asciz
+
+CPOPJ:	POPJ P,
+
+feep:	asciz/** /
+CRLF:	asciz/
+/
+
+CMACMA:	ASCIZ/,,/
+
+SEP:	ASCIZ/-----
+/
+
+bletch: asciz\
+WRONG SYSTEM VERSION - REPURIFY
+\
+
+MEMHOL:	ASCIZ\
+WARNING: THERE IS A HOLE IN MEMORY
+\
+
+	loc 164
+
+STACK==100		;DEFAULT TO 100 WORDS
+pdl:	BLOCK STACK
+
+	loc 264		; not sure why removing this causes errors
+
+sysmsg: move .cf.,[-stack,,pdl-1]	; setup current stack frame pointer
+	move p,[-stack,,pdl-1]		; setup current PDL pointer
+
+	.call [setz ? sixbit/OPEN/ ? [.uao,,tyoc] 
+	  [sixbit/DSK/] ? [sixbit/CDATA/] ? [sixbit/>/] ? setz [sixbit/DRAGON/]]
+	  .value
+
+	pushj p,ritual		; assurance of purity
+	movei count,1
+	lsh count,0		; instruction patched through abstb1 (SYSMLN)
+	movei pt,0              ; instruction patched through abstb1 (SYSMBF)
+	jrst papsav
+
+g00002:
+	skipn a,(pt)
+	  jrst nxloop
+	jumpge a,g00002+5
+	movei t1,[feep]
+	pushj p,type
+	hllz f,.prsva
+	hrli a,440700
+	movei c,t1(pt)
+loop1:
+	ildb b,.prsva
+	jumpe b,g00013
+	caie b,"~
+	jrst loop1+6
+	pushj p,argot
+	jrst g00011
+	tyo b
+g00011:
+	jrst loop1
+g00013:
+	tlnn f,gotit+4
+	  jrst g00014
+	tyo [" ]
+	pushj p,argot
+	jrst g00013
+g00014:
+	movei t1,[crlf]
+	pushj p,type
+nxloop:
+	addi pt,4
+	sojg count,g00002
+g00003:
+	movei t1,[sep]
+	pushj p,type
+
+papsav:	movei c,0		; instrucxtion patched by abstb1 (SYSCON)
```

Comment:
instrucxtion -> instruction.

Reply:
I’ve fixed this in my CALRET version. Will submit a PR when I have it comparing equally.